clouddriver-google-common/src/main/groovy/com/netflix/spinnaker/clouddriver/googlecommon/deploy/GoogleApiException.java

```diff
@@ -19,40 +19,61 @@
 import com.google.api.client.googleapis.json.GoogleJsonError;
 import com.google.api.client.googleapis.json.GoogleJsonError.ErrorInfo;
 import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
 import java.util.Optional;
+import lombok.Value;
 
+@NonnullByDefault
 public class GoogleApiException extends RuntimeException {
   GoogleApiException(String message) {
     super(message);
   }
 
   static GoogleApiException fromGoogleJsonException(GoogleJsonResponseException e) {
-    Optional<ErrorInfo> optionalErrorInfo =
-        Optional.ofNullable(e.getDetails()).map(GoogleJsonError::getErrors)
-            .orElse(ImmutableList.of()).stream()
-            .findFirst();
-
-    if (optionalErrorInfo.isPresent()) {
-      ErrorInfo errorInfo = optionalErrorInfo.get();
-      String message =
-          String.format(
-              "Operation failed. Last attempt returned status code %s with error message %s and reason %s.",
-              e.getStatusCode(), errorInfo.getMessage(), errorInfo.getReason());
-      if ("resourceInUseByAnotherResource".equals(errorInfo.getReason())) {
-        return new ResourceInUseException(message);
+    ErrorDetails errorDetails = ErrorDetails.fromGoogleJsonException(e);
+    if (errorDetails.getReason().equals("resourceInUseByAnotherResource")) {
```

Comment:
Good catch! I thought I had made this null-safe because we're never passing in null, but I forgot that we're passing `errorDetails.getMessage()` (and similar for reason) to the constructor and they can return null.

I've fixed it by just explicitly writing out the constructor and doing the `nullToEmpty` there rather than marking it nullable, so the calling code doesn't need to worry about nulls.